service/create/service.go

```diff
@@ -824,7 +842,7 @@ func (s *Service) addFunc(obj interface{}) {
 	// If the policy couldn't be created and some instances didn't exist before, that means that the cluster
 	// is inconsistent and most problably its deployment broke in the middle during the previous run of
 	// aws-operator.
-	if anyMastersCreated && (kmsKeyErr != nil || policyErr != nil) {
+	if anyMastersCreated && (kmsKeyErr != nil || masterPolicyErr != nil || workerPolicyErr != nil) {
```

Comment:
There is an issue for whether this check is still needed. I added the new errors here for completeness.

https://github.com/giantswarm/aws-operator/issues/310